WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/logic/interval/IntervalManagerBase.java

```diff
@@ -73,4 +75,21 @@ public IntervalBase getIntervalOfType(IntervalType type) {
 		}
 		return collectedIntervals;
 	}
+
+	/**
+	 * @return An {@link ArrayList} of intervals of the specified Perspective
+	 *         type.
+	 */
+	protected List<IntervalBase> getPerspectiveIntervalsOfType(Perspective type) {
```

Comment:
What about writing a generic function getIntervalsOfType that is private and called by getEditorIntervalsOfDocType and getPerspectiveIntervalsOfType to provide typed versions?